### PR TITLE
Fix zero defaulting issue for ResourceQuotas

### PR DESCRIPTION
**Pull request by pkprzekwas**
**What this PR does / why we need it**:
Setting default values to zero blocks cluster creation if any of quota parameters is omitted in API request.

**What type of PR is this?**
/kind bug

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/ee/resource-quota/handler.go [hunk 1]
```diff
@@ -375,5 +375,17 @@ func convertToCRDQuota(quota apiv2.Quota) (kubermaticv1.ResourceDetails, error)
...
+		resourceDetails.CPU = nil
+	}
+	if resourceDetails.Memory.IsZero() {
+		resourceDetails.Memory = nil
+	}
+	if resourceDetails.Storage.IsZero() {
+		resourceDetails.Storage = nil
+	}
+
+	return *resourceDetails, nil
```
Comment (ahmedwaleedmalik) [inline pkg/ee/resource-quota/handler.go]: This seems a bit hackish to me. Shouldn't we just treat 0 as nil? For example, If I create the resource quota directly via CR and give `CPU` the value `0`. Then I'll end up with the same issue, right?

Comment (pkprzekwas) [reply pkg/ee/resource-quota/handler.go]: > If I create the resource quota directly via CR and give CPU the value 0. Then I'll end up with the same issue, right?

Yes. It actually may be a good idea as there is no value in distinguishing between `0` and `nil` unless we want to allow for blocking resource creation inside the project. But at this stage I don't want to risk changing CR and the way controllers interpret zero value  (cc @lsviben )

I'm currently testing changing values to pointers in the API model and replacing PATCH with PUT as an alternative solution.

Comment (ahmedwaleedmalik) [reply pkg/ee/resource-quota/handler.go]: I agree with you, for now, let's just settle with this.

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

File: pkg/ee/resource-quota/handler.go (hunk 1 above)
Comment (pkprzekwas) [reply pkg/ee/resource-quota/handler.go]: I've replaced values with pointers and to make the API consistent with the behaviour under the hood introduced PUT in place of PATCH. I'll get rid of PATCH soon, but first I want to update the Dashboard, hence it got deprecated for now. Please take a look @ahmedwaleedmalik if possible.

Comment (pkprzekwas) [reply pkg/ee/resource-quota/handler.go]: oh, I see you already taken a glance :D

Comment (ahmedwaleedmalik) [reply pkg/ee/resource-quota/handler.go]: Yes :D

**Pull request merged**